src/java/com/puppetlabs/jruby_utils/jruby/InternalScriptingContainer.java

```diff
@@ -10,12 +21,150 @@
 public class InternalScriptingContainer
         extends org.jruby.embed.ScriptingContainer
         implements ScriptingContainer {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(
+            InternalScriptingContainer.class);
+
+    /**
+     * {@link #terminateJarIndexCacheEntries(URL[])} method for more info on the
+     * purpose of the jar index cache purge code
+     */
+    private static Map<String, Object> jarIndexCache = null;
+    private static Method jarIndexReleaseMethod = null;
+    private static Exception jarIndexCacheException = null;
+    private static Method classLoaderTempDirMethod = null;
+
+    /**
+     * We have to use reflection to get access to the jar index cache since it
+     * is not public.
+     */
+    static {
+        try {
+            Class jarResourceClass = Class.forName("org.jruby.util.JarResource");
+            Field jarCacheField = jarResourceClass.getDeclaredField("jarCache");
+            jarCacheField.setAccessible(true);
+            Object jarCache = jarCacheField.get(jarResourceClass);
+            Class jarCacheClass = jarCache.getClass();
+            Field indexCacheField = jarCacheClass.getDeclaredField("indexCache");
+            indexCacheField.setAccessible(true);
+
+            @SuppressWarnings("unchecked") Map<String, Object> indexCache =
+                    (Map<String, Object>) indexCacheField.get(jarCache);
+            jarIndexCache = indexCache;
+
+            Class<?> jarIndexClass = Class.forName("org.jruby.util.JarCache$JarIndex");
```

Comment:
It's the declared return type for the [Class.forName()](https://docs.oracle.com/javase/7/docs/api/java/lang/Class.html#forName(java.lang.String,%20boolean,%20java.lang.ClassLoader)) method.  The <> part indicates that is a generic type.  The ? inside of it would signify that basically that the return type is expected to be a Class of a type extending from an `Object`.  I might have been able to do this as a `Class<JarIndex>` instead but `JarIndex` isn't public.